scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt

```diff
@@ -10,22 +10,53 @@ import android.os.Bundle
 import com.tinder.scarlet.Lifecycle
 import com.tinder.scarlet.ShutdownReason
 import com.tinder.scarlet.lifecycle.LifecycleRegistry
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
 
 internal class ApplicationResumedLifecycle(
     application: Application,
     private val lifecycleRegistry: LifecycleRegistry
 ) : Lifecycle by lifecycleRegistry {
+    private val job = SupervisorJob()
+    private var scope = CoroutineScope(job + Dispatchers.IO)
 
     init {
         application.registerActivityLifecycleCallbacks(ActivityLifecycleCallbacks())
     }
 
     private inner class ActivityLifecycleCallbacks : Application.ActivityLifecycleCallbacks {
-        override fun onActivityPaused(activity: Activity) = lifecycleRegistry.onNext(
-            Lifecycle.State.Stopped.WithReason(ShutdownReason(1000, "App is paused"))
-        )
+        var isResumed: Boolean = false
+        var job: Job? = null
 
-        override fun onActivityResumed(activity: Activity) = lifecycleRegistry.onNext(Lifecycle.State.Started)
+        override fun onActivityPaused(activity: Activity) {
+            isResumed = false
+
+            job = scope.launch {
+                supervisorScope {
```

Comment:
we don't need an additional `SupervisorScope`.

Reply:
That's true 👌